clustermesh/clustermesh.go

```diff
@@ -2243,7 +2272,7 @@ func (k *K8sClusterMesh) DisconnectWithHelm(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-	if err = k.validateInfoForConnect(aiLocal, aiRemote); err != nil {
```

Comment:
I think rather than creating a new validation function, we can safely remove this validation from disconnect. The validation in `validateInfoForConnect` is there to prevent creating a broken clustermesh configuration, so we shouldn't need to re-validate that before removing the configuration as well.

The reason why this is breaking is because we're using `shallowExtractAccessInfo` rather than the full `extractAccessInfo`, but using the same validate function (which is assuming all of the data has been extracted).

Reply:
@thorn3r do you have news?